src/test/java/com/commercetools/project/sync/shoppinglist/ShoppingListSyncerTest.java

```diff
@@ -0,0 +1,110 @@
+package com.commercetools.project.sync.shoppinglist;
+
+import static io.sphere.sdk.json.SphereJsonUtils.readObjectFromResource;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.commercetools.sync.shoppinglists.ShoppingListSync;
+import com.commercetools.sync.shoppinglists.utils.ShoppingListReferenceResolutionUtils;
+import io.sphere.sdk.client.SphereApiConfig;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.queries.PagedQueryResult;
+import io.sphere.sdk.shoppinglists.ShoppingList;
+import io.sphere.sdk.shoppinglists.ShoppingListDraft;
+import io.sphere.sdk.shoppinglists.queries.ShoppingListQuery;
+import java.time.Clock;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import uk.org.lidalia.slf4jtest.LoggingEvent;
+import uk.org.lidalia.slf4jtest.TestLogger;
+import uk.org.lidalia.slf4jtest.TestLoggerFactory;
+
+class ShoppingListSyncerTest {
+  private final TestLogger syncerTestLogger =
+      TestLoggerFactory.getTestLogger(ShoppingListSyncer.class);
+
+  @BeforeEach
+  void setup() {
+    syncerTestLogger.clearAll();
+  }
+
+  @Test
+  void of_ShouldCreateShoppingListSyncerInstance() {
+    // test
+    final ShoppingListSyncer shoppingListSyncer =
+        ShoppingListSyncer.of(
+            mock(SphereClient.class), mock(SphereClient.class), mock(Clock.class));
+
+    // assertion
+    assertThat(shoppingListSyncer).isNotNull();
+    assertThat(shoppingListSyncer.getSync()).isInstanceOf(ShoppingListSync.class);
+  }
+
+  @Test
+  void transform_ShouldReplaceShoppingListReferenceIdsWithKeys() {
+    // preparation
+    final ShoppingListSyncer shoppingListSyncer =
+        ShoppingListSyncer.of(
+            mock(SphereClient.class), mock(SphereClient.class), mock(Clock.class));
+    final List<ShoppingList> shoppingList =
+        Collections.singletonList(readObjectFromResource("shopping-list.json", ShoppingList.class));
+
+    // test
+    final CompletionStage<List<ShoppingListDraft>> draftsFromPageStage =
+        shoppingListSyncer.transform(shoppingList);
+
+    // assertion
+    assertThat(draftsFromPageStage)
+        .isCompletedWithValue(
+            ShoppingListReferenceResolutionUtils.mapToShoppingListDrafts(shoppingList));
+  }
+
+  @Test
+  void getQuery_ShouldBuildShoppingListQuery() {
+    // test
+    final ShoppingListSyncer shoppingListSyncer =
+        ShoppingListSyncer.of(
+            mock(SphereClient.class), mock(SphereClient.class), mock(Clock.class));
+
+    // assertion
+    final ShoppingListQuery query = shoppingListSyncer.getQuery();
+    assertThat(query).isEqualTo(ShoppingListReferenceResolutionUtils.buildShoppingListQuery());
```

Comment:
The assertion does not give any hint, instead, you are testing the internal parts of the method 🤔

Reply:
Removed internal method call and created expected query in the test class for asserted. https://github.com/commercetools/commercetools-project-sync/pull/200/commits/cf9593930e476b39ee09a6d568f717e71f5601ea

Reply:
Not sure what you are doing in this one line and lots of casts, maybe check the test, below:

```` java
assertThat(buildShoppingListQuery().expansionPaths())
            .containsExactly(
                ExpansionPath.of("customer"),
                ExpansionPath.of("custom.type"),
                ExpansionPath.of("lineItems[*].variant"),
                ExpansionPath.of("lineItems[*].custom.type"),
                ExpansionPath.of("textLineItems[*].custom.type"));
    }
````

Reply:
Corrected, https://github.com/commercetools/commercetools-project-sync/pull/200/commits/b2e2c2b89fa037fadfe356320e5df1d832d3765a